src/tls/Makefile.common

```diff
@@ -14,48 +35,79 @@ INCLUDE_PATHS= \
 	$(HACL_HOME)/code/lib \
 	$(KREMLIN_HOME)/kremlib \
 	$(MITLS_HOME)/libs/ffi \
-	./concrete-flags \
-        ./concrete-flags/$(FLAVOR)$(CONCRETE_FLAVOR)
+	$(SPECINC)
 
 ifeq (,$(MITLS_NODEBUG))
   INCLUDE_PATHS += ./concrete-flags/Debug
 else
   INCLUDE_PATHS += ./concrete-flags/NoDebug
 endif
 
-CACHE_DIR=cache.$(FLAVOR)$(CONCRETE_FLAVOR)
-
 FSTAR=$(FSTAR_HOME)/bin/fstar.exe --cache_dir $(CACHE_DIR) --cache_checked_modules $(OTHERFLAGS)
 
-EXTRACT_DIR=extract/$(FLAVOR)$(CONCRETE_FLAVOR)
+VFLAGS?=--use_hints --use_hint_hashes --record_hints --warn_error -271
```

Comment:
Add a `HINTS_DIR` variable and `--hint_file $(HINTS_DIR)/$<.hints` to avoid polluting the main directory?